simple/utils/spectral_types.py

```diff
@@ -3,320 +3,334 @@
-
-        spectral_type_codes.append(spt_code)
-    return spectral_type_codes
 
-
-def convert_spt_code_to_string_to_code(spectral_codes, decimals=1):
+    logger.debug(f"Trying to convert: `{spectral_type_string}`")
+    spt_code = np.nan
+    if spectral_type_string == "":
+        logger.debug("Empty spectral_type_string")
+        return None
+    if spectral_type_string == "null":
+        return None
+    # identify main spectral class, loop over any prefix text to identify MLTY
+    for i, item in enumerate(spectral_type_string):
+        if item == "M":
+            spt_code = 60
+            break
+        elif item == "L":
+            spt_code = 70
+            break
+        elif item == "T":
+            spt_code = 80
+            break
+        elif item == "Y":
+            spt_code = 90
+            break
+        else:  # only trigger if not MLTY
+            i = 0
+    # find integer or decimal subclass and add to spt_code
+    if re.search(r"\d*\.?\d+", spectral_type_string[i + 1 :]) is None:
+        spt_code = spt_code
+    else:
+        spt_code += float(re.findall(r"\d*\.?\d+", spectral_type_string[i + 1 :])[0])
+
+    return spt_code
+
+
+def convert_spt_code_to_string(spectral_code, decimals=1):
     """
     Convert spectral type codes to string values
 
     Parameters
     ----------
-    spectral_codes : list[float]
-        List of spectral type codes
+    spectral_code : float
+        A spectral type code
+
+    decimals : int
+        Number of decimal places to include in the spectral type string
 
     Returns
     -------
-    spectral_types : list[str]
-        List of spectral types
+    spectral_type_string : str
+        spectral type string
     """
-    if isinstance(spectral_codes, float):
-        spectral_codes = [spectral_codes]
-
-    spectral_types = []
-    for spt in spectral_codes:
-        spt_type = ""
-
-        # Identify major type
-        if 60 <= spt < 70:
-            spt_type = "M"
-        elif 70 <= spt < 80:
-            spt_type = "L"
-        elif 80 <= spt < 90:
-            spt_type = "T"
-        elif 90 <= spt < 100:
-            spt_type = "Y"
-
-        # Numeric part of type
-        format = f".{decimals}f"
-        spt_type = f"{spt_type}{spt % 10:{format}}"
-        logger.debug(f"Converting: {spt} -> {spt_type}")
-
-        spectral_types.append(spt_type)
-
-    return spectral_types
+    spt_type = ""
+
+    # Identify major type
+    if 60 <= spectral_code < 70:
+        spt_type = "M"
+    elif 70 <= spectral_code < 80:
+        spt_type = "L"
+    elif 80 <= spectral_code < 90:
+        spt_type = "T"
+    elif 90 <= spectral_code < 100:
+        spt_type = "Y"
+
+    # Numeric part of type
+    format = f".{decimals}f"
+    spt_type = f"{spt_type}{spectral_code% 10:{format}}"
+    logger.debug(f"Converting: {spectral_code} -> {spt_type}")
+
+    return spt_type
+
+
+def adopt_spectral_type(db, source, spectral_type_error):
```

Comment:
These new adoption functions are un-tested, but I think we can take care of that when we address https://github.com/SIMPLE-AstroDB/SIMPLE-db/issues/542

Reply:
added #TODOs in the test file